R/method-filterFeatures.R

```diff
@@ -0,0 +1,597 @@
+#'  be used to flag features that result from contamination in the solvent of
+#'  the samples. A new column `possible_contaminant` is added to the
+#'  `featureDefinitions` reflecting this.
+#'
+#' For specific examples, see the help pages of the individual parameter classes
+#' listed above.
+#'
+#' @param object `XcmsExperiment` or `SummarizedExperiment`. For an
+#' `XcmsExperiment` object, the `featureValues(object)` will be evaluated, and
+#' for `Summarizedesxperiment` the `assay(object, assay)`. The object will be
+#' filtered.
+#'
+#' @param filter The parameter object selecting and configuring the type of
+#' filtering. It can be one of the following classes: [`RsdFilter`],
+#' [`DratioFilter`], [`PercentMissingFilter`] or [`BlankFlag`].
+#'
+#' @param assay For filtering of `SummarizedExperiment` objects only. Indicates
+#' which assay the filtering will be based on. Note that the features for the
+#' entire object will be removed, but the computations are performed on a single
+#' assay. Default is 1, which mean the first assay of the `object` will
+#' be evaluated.
+#'
+#' @param ... Optional parameters.
+#'
+#' @name filterFeatures
+#'
+#' @author Philippine Louail
+#'
+#' @importFrom MetaboCoreUtils rowRsd rowDratio rowPercentMissing rowBlank
+#'
+#' @importMethodsFrom ProtGenerics filterFeatures
+#'
+#' @references
+#'
+#' Broadhurst D, Goodacre R, Reinke SN, Kuligowski J, Wilson ID, Lewis MR,
+#' Dunn WB. Guidelines and considerations for the use of system suitability
+#' and quality control samples in mass spectrometry assays applied in
+#' untargeted clinical metabolomic studies. Metabolomics. 2018;14(6):72.
+#' doi: 10.1007/s11306-018-1367-3. Epub 2018 May 18. PMID: 29805336;
+#' PMCID: PMC5960010.
+#'
+#' @examples
+#' ## See the vignettes for more detailed examples
+#' library(MsExperiment)
+#'
+#' ## Load a test data set with features
+#' test_xcms <- loadXcmsData()
+#'
+#' ## Set up parameter to filter based on coefficient of variation
+#' rsd_filter <- RsdFilter(qcIndex = sampleData(test_xcms)$sample_type == "QC")
+#'
+#' filtered_data_rsd <- filterFeatures(object = test_xcms, filter = rsd_filter)
+#'
+#' ## Set up parameter to filter based on D-ratio and filter
+#' dratio_filter <- DratioFilter(qcIndex = sampleData(test_xcms)$sample_type == "QC",
+#'                               studyIndex = sampleData(test_xcms)$sample_type == "study")
+#'
+#' filtered_data_dratio <- filterFeatures(object = test_xcms,
+#' filter = dratio_filter)
+#'
+#' ## Set up parameter to filter based on the percent of missing data
+#' missing_data_filter <- PercentMissingFilter(f = sampleData(test_xcms)$sample_type)
+#'
+#' filtered_data_missing <- filterFeatures(object = test_xcms,
+#' filter = missing_data_filter)
+#'
+#' ## Set up parameter to flag possible contaminant based on blank samples
+#' filter <- BlankFlag(
+#'     qcIndex = sampleData(test_xcms)$sample_type == "QC",
+#'     blankIndex = sampleData(test_xcms)$sample_type == "study")
```

Comment:
here i'm not sure what to do, I think it might be confusing for people that i use study samples as blank samples(because the object does not have blanks). I can explain it but is it a good idea ? 
Otherwise should I had like 2 randomly generated blank sample to the object and use that as an example ?